PyPDF2/pagerange.py

```diff
@@ -170,4 +170,4 @@ def parse_filename_page_ranges(
     return pairs
 
 
-PageRangeSpec = Union[str, PageRange, Tuple[int, int], Tuple[int, int, int]]
+PageRangeSpec = Union[str, PageRange, Tuple[int, int], Tuple[int, int, int], List[int]]
```

Comment:
This is a change that is not related to cloning, right? It was just wrong in the tests before.

As this PR is rather big and might take a while to review, I would like to move this to a separate PR ("Miscellaneous fixes" or similar)

Reply:
code was quite often too tight. For this it's OK will do